src/scaffoldmaker/meshtypes/meshtype_3d_solidcylinder1.py

```diff
@@ -0,0 +1,127 @@
+
+
+class MeshType_3d_solidcylinder1(Scaffold_base):
+    '''
+    classdocs
+    '''
+    @staticmethod
+    def getName():
+        return '3D Solid Cylinder 1'
+
+    @staticmethod
+    def getDefaultOptions(parameterSetName='Default'):
+        return {
+            'Cylinder height' : 5.0,
+            'Major radius' : 2.5,
+            'Minor radius' : 1.0,
+            'Radius reduction rate' : 0.08,
+            'Number of elements across' : 8,
+            'Number of elements up' : 5,
+            'Number of elements along' : 5,
+            'Use cross derivatives' : False,
+            'Refine' : False,
+            'Refine number of elements along' : 1,
+            'Refine number of elements across' : 1,
+            'Refine number of elements up' : 1
+        }
+
+    @staticmethod
+    def getOrderedOptionNames():
+        return [
+            'Cylinder height',
+            'Major radius',
+            'Minor radius',
+            'Radius reduction rate',
+            'Number of elements across',
+            'Number of elements up',
+            'Number of elements along',
+            'Use cross derivatives',
+            'Refine',
+            'Refine number of elements along',
+            'Refine number of elements across',
+            'Refine number of elements up'
+        ]
+
+    @staticmethod
+    def checkOptions(options):
+        for key in [
+            'Number of elements along',
+            'Number of elements across',
+            'Refine number of elements up',
+            'Refine number of elements along']:
+            if options[key] < 1:
+                options[key] = 1
+        # if (options['Number of elements through wall'] < 2) :
+        #     options['Number of elements through wall'] = 1
+        # if (options['Number of elements around'] < 2) :
+        #     options['Number of elements around'] = 2
+
+
+    @staticmethod
+    def generateBaseMesh(region, options):
+        """
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: None
+        """
+        height = options['Cylinder height']
+        majorRadius = options['Major radius']
+        minorRadius = options['Minor radius']
+        rate = options['Radius reduction rate']
+        elementsCountAcross = options['Number of elements across']
+        elementsCountUp = options['Number of elements up']
+        elementsCountAlong = options['Number of elements along']
+        useCrossDerivatives = options['Use cross derivatives']
+
+        fm = region.getFieldmodule()
+        fm.beginChange()
+        coordinates = findOrCreateFieldCoordinates(fm)
+
+        axis1 = [1.0, 0.0, 0.0]
+        axis2 = [0.0, 1.0, 0.0]
+        axis3 = [0.0, 0.0, 1.0]
+
+        fm = region.getFieldmodule()
+        fm.beginChange()
```

Comment:
These two lines should not have been repeated -- caused change caching to remain active which reduced performance.

Reply:
Done